source/ota.c

```diff
@@ -1265,6 +1013,78 @@ static OtaErr_t jobNotificationHandler( const OtaEventData_t * pEventData )
     return ( OTA_SignalEvent( &eventMsg ) == true ) ? OTA_ERR_NONE : OTA_ERR_EVENT_Q_SEND_FAILED;
 }
 
+static void freeFileContextMem( OtaFileContext_t * const pFileContext )
+{
+    if( pFileContext != NULL )
+    {
+        /* Free or clear the filepath buffer.*/
+        if( ( pFileContext->pFilePath != NULL ) && ( pFileContext->filePathMaxSize > 0 ) )
```

Comment:
It appears in all of these conditionals that a somethingMaxSize of 0 is supposed to indicate heap was used.  Still, wouldn't free be called here is the pointer == NULL ?

Reply:
Std free called with NULL should not harm , but I have updated the if else conditions in 7d00faf